### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by ab-noori**
### 🌟Features:

> - [ ] - `Nameable` interface with correct_name method is implemented.
> - [ ] - `Person` class is refactored to inherit from `Nameable`.
> - [ ] - `Base Decorator` class is implemented.
> - [ ] - `CapitalizeDecorator` and `TrimmerDecorator` classes are implemented.

### CODE REVIEW

File: Decorator.rb
Comment (fobadara) [inline Decorator.rb:1]: [REQUIRED] Decorator should be a class that inherits from Nameable and in the. constructor a namable object from params is assigned to an instance variable. You should also have a correct_name method that returns the result of the correct_name method of @namable

File: Person.rb [hunk 1]
```diff
@@ -8,18 +11,14 @@ def initialize(name = "Unknown", age = 0, parent_permission = true)
...
-  def age=(age)
-    @age = age
-  end
-
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name 
+    @name
```
Comment (fobadara) [inline Person.rb:19]: [REQUIRED] The correct_name method should look something like this:
```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

File: CapitalizeDecorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative "BaseDecorator.rb"
+
+class CapitalizeDecorator < BaseDecorator
+    def correct_name
+        super.capitalize
+    end
```
Comment (fobadara) [inline CapitalizeDecorator.rb]: [REQUIRED] Your correct_name should look something like this:
```ruby
  def correct_name
      @nameable.correct_name.capitalize
    end
```

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @ab-noori, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Good first attempt
✅ Good file structure


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Decorator.rb
Comment (ab-noori) [reply Decorator.rb:1]: **Hi @fobadara 👋,**
Thank you so much 🙏  for reviewing my project.

- [ ] Kindly check the `BaseDecorator` class. I have implemented your suggestion. The `Decorator` file is just for testing the result.

File: Person.rb (hunk 1 above)
Comment (ab-noori) [reply Person.rb:19]: Your nice feedback is really appreciated 🙏.

- [ ] I have checked, and made it fix 🎯. The changes applied to the `TrimmerDecorator` class.

File: CapitalizeDecorator.rb (hunk 2 above)
Comment (ab-noori) [reply CapitalizeDecorator.rb]: Your nice feedback is really appreciated 🙏.

- [ ] I have checked, and made it fix 🎯. The changes applied to the `CapitalizeDecorator` class.

**Review by joel-kalema — APPROVED** [review]
Hi @ab-noori 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
 You have a good file structure 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @joel-kalema 👋,**
Thank you so much 🙏  for reviewing and approving my project.
Your nice feedback is really appreciated 🙏.

Thank-You-Very-Much

**Pull request merged**